CodeEditModules/Modules/WorkspaceClient/src/Model/FileItem.swift

```diff
@@ -257,7 +257,11 @@ public extension WorkspaceClient {
             var fileUrl = self.url
             while FileItem.fileManger.fileExists(atPath: fileUrl.path) {
                 let previousName = fileUrl.lastPathComponent
-                fileUrl = fileUrl.deletingLastPathComponent().appendingPathComponent("\(previousName) copy")
+                let filextension = fileUrl.pathExtension
+                let fileName = filextension.isEmpty ?
+                previousName : previousName.replacingOccurrences(of: ".\(filextension)", with: "")
+                fileUrl = fileUrl.deletingLastPathComponent()
+                fileUrl.appendPathComponent("\(fileName) copy.\(filextension)")
```

Comment:
somehow i would split this also in a `appendPathCompontent(newName)` and a `. appendPathExtension(fileextensiont)`

Reply:
you should also consider there are files existing without file extension: for example Dockerfile